### DISCUSSION TITLE
add new property to  support selectionOnDrag like reactflow

### DISCUSSION THREAD

**Discussion by Deuscx** [bcakmakoglu/vue-flow#1676] [Ideas]
> If you prefer figma/sketch/design tool controls you can set panOnScroll={true} and selectionOnDrag={true}:
> 
> pan: Space + drag mouse, scroll, middle or right mouse
> zoom: pitch or cmd + scroll
> create selection: drag mouse

https://reactflow.dev/learn/concepts/the-viewport

like panOnDrag ,return the selectionOnDrag in useVueflow

```
const { selectionOnDrag } = useVueflow()
```

and I found that the selectionOnDrag prop is removed in `1.9.0` version. How to achieve functions similar to Figma?

![image](https://github.com/user-attachments/assets/7e58906d-f6df-46bd-8c2c-91d487722917)

**Comment by bcakmakoglu** [comment]
This is already possible by simply assigning `:selectionKeyCode="true"`.
A new prop is not necessary at the moment.

**Reply by 799189288** [reply]
there is a problem, if i both set selectionKeyCode and pan-activation-key-code: Space. and when i press the space key i can pan and select at the same time. it's so weired.
what i want is when i press the space the select behavior will not happen.

**Reply by bcakmakoglu** [reply]
Then assign a different key to selectionKeyCode? 😄

**Reply by 799189288** [reply]
> Then assign a different key to selectionKeyCode? 😄

I am developing a Vue version of the Dify platform. The left toolbar features two mode switches: one for pan mode and one for selection mode. Additionally, it must support pan with the spacebar. (Pressing the spacebar does not switch modes; it simply enters pan mode.) This is how I use it
```
<VueFlow
        :nodes="nodes"
        :edges="edges"
        :node-types="nodeTypes"
        :edge-types="edgeTypes"
        :default-viewport="{ zoom: 1, x: 0, y: 0 }"
        :min-zoom="0.25"
        :max-zoom="2"
        :zoom-on-scroll="true"
        :zoom-on-pinch="true"
        :zoom-on-double-click="true"
        :pan-on-scroll="controlMode === 'pointer' && !workflowReadOnly"
        :pan-on-drag="controlMode === 'hand' || false"
        :selection-key-code="controlMode === 'pointer' && !workflowReadOnly"
        :nodes-draggable="!nodesReadOnly"
        :nodes-connectable="!nodesReadOnly"
        :nodes-focusable="!nodesReadOnly"
        :edges-focusable="!nodesReadOnly"
        :connection-line-component="CustomConnectionLine"
        :connection-line-style="{ zIndex: 1001 }"
        :pan-activation-key-code="['Space']"
></VueFlow>
```
As you suggested, I might need to monitor keyboard keystrokes additionally, rather than simply binding them on VueFlow。But it seems that ReactFlow hasn't done this.

**Reply by bcakmakoglu** [reply]
>But it seems that ReactFlow hasn't done this.

I have no idea what you mean exactly, you haven't explained your problem or your use-case properly to me - at least I haven't understood at all what your issue is besides binding the same key to two different actions 🤔

**Reply by 799189288** [reply]
> > But it seems that ReactFlow hasn't done this.
> 
> I have no idea what you mean exactly, you haven't explained your problem or your use-case properly to me - at least I haven't understood at all what your issue is besides binding the same key to two different actions 🤔

Perhaps VueFlow can handle these internally, while ReactFlow can achieve this effect by just setting two properties, without any conflicts
```
   panOnDrag={controlMode === ControlMode.Hand || [1]}
   selectionOnDrag={controlMode === ControlMode.Pointer && !workflowReadOnly}
  ```

**Reply by bcakmakoglu** [reply]
Still no idea what your issue is exactly besides binding the same key to two actions.
Your messages are vague and I am not able to deduce what you're trying to tell me.

You want to press space to drag, so you set `panActivationKeyCode` to `Space`, that looks fine.
You don't want selections on space (the default key for this isn't even `Space` to begin with, it's `Shift`), so don't set a selectionKeyCode to `Space`. If you want the `selectionOnDrag` behavior, set the keycode to `true`.

What exactly is the problem here?

**Reply by 799189288** [reply]
the problem is, how can i set selectionOnDrag to true when i press the space? Do I have to additionally monitor keyboard events. like space key down.

**Reply by bcakmakoglu** [reply]
You don't need "selectionOnDrag" if you want to use a key to activate selection, does two don't make sense together.
Either you assert that selections are active without a key (which is what `selectionOnDrag` in RF does, no key activation whatsoever necessary) or you assert that a key activation + drag is necessary (which is the default behavior).

You are trying to assert multiple things at once that don't make any sense, if you set `panActivationKeyCode` to `Space` (the default key) it will activate when you press space, regardless of what other features you assigned to the same key.
Drop the `panActivationKeyCode` entirely (by setting it to `null` for example) if you want to use `Space` as your selection key code or assign it to a different key.

**Reply by 799189288** [reply]
1. Currently, I want to be able to drag the mouse to move the canvas when I press the space bar. However, when I don't press the space bar, vue-flow should determine whether to select the mode or the drag-and-move mode based on my variable controlMode. controlMode has two values: "Pointer" and "Hand"
2. vue-flow currently only allows enabling selection mode by setting selectionKeyCode to true, and does not have a selectionOnDrag parameter
3. So when the variable controlMode === Pointer, the value of selectionKeyCode is true. When I press the space bar, the expected effect is that （the variable controlMode remains unchanged, but vueflow becomes draggable.）
4. So when I set the following parameters, 
`
:pan-activation-key-code="['Space']"
:pan-on-drag="controlMode === 'hand' || false"
:selection-key-code="controlMode === 'pointer' && !workflowReadOnly"
`
a conflict arose, didn't it? The simultaneous activation of selectionKeyCode and pan-activation-key-code led to this conflict. forcing me to set selectionKeyCode to true to enable the selection mode. This is unlike RF's selectionOnDrag

**Reply by bcakmakoglu** [reply]
Sorry, I still don't understand what the *issue* is supposed to be.

> forcing me to set selectionKeyCode to true to enable the selection mode. This is unlike RF's selectionOnDrag

You would have to set `selectionOnDrag` to `true` in RF, while in VF you set the keycode to `true` - still not seeing what the issue is.

**Reply by 799189288** [reply]
Reactflow official website has a example, [example](https://reactflow.dev/learn/concepts/the-viewport#design-tool-viewport-controls) called Design tool viewport controls. 
Do you have any doubts about the questions I asked

**Reply by bcakmakoglu** [reply]
Here you go: https://codesandbox.io/p/devbox/youthful-moon-ry9869?file=%2Fsrc%2FApp.vue%3A7%2C66&workspaceId=ws_7iZPUfEyQC7Podbs7wwbxV

Same example as the one you linked, still not getting what your issue supposedly is ^^

**Reply by 799189288** [reply]
try to press the spacebar and u will see the difference.

**Reply by bcakmakoglu** [reply]
Alright, I see what you mean. That's a fixable bug, I'll add it to the next release :)

**Reply by bcakmakoglu** [reply]
Should be resolved with `1.48.2` - pull the latest version and see if that resolves your problem ^^

**Reply by 799189288** [reply]
Thanks, the issue has been fixed
